compiler/src/main/kotlin/ast/ast.kt

```diff
@@ -114,14 +108,23 @@ data class StateDecl(
     val body: List<Stmt>
 ) : Decl()
 
-data class Coordinate(val axes: List<Int>)
+data class Coordinate(val ctx: ParseTree, val axes: List<Int> = emptyList())
```

Comment:
```suggestion
data class Coordinate(val ctx: ParseTree, val coordX: Int, val coordY: Int?)
```

Reply:
Having the axes as a list allows for easier data manipulation and validation, and have proven useful to me